.github/workflows/send-notification.yml

```diff
@@ -0,0 +1,75 @@
+#
+#      jobs:
+#         notification:
+#           name: Notify about build failures
+#           # List the other build jobs that we want to "monitor for failures":
+#           needs: build
+#           if: ${{failure()}}
+#           uses: hibernate/.github/.github/workflows/send-notification.yml@main
+#         build:
+#           name: ...
+#           ...
+#   2. Done ;)
+
+name: 'Send Zulip notification'
+on:
+  workflow_call:
+permissions:
+  contents: read
+jobs:
+  notify:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Figure out where to send the notification to
+        id: notification-destination
+        shell: bash
+        run: |
+          case "${{ github.repository }}" in
+            # Let's try to figure out where we want to send the message to
+            #  without requiring the input params, so that in the future we wouldn't need
+            #  to go to all the repos using this one and update them if we decide to change things up a bit.
+            "hibernate/hibernate-validator")
+              echo "stream=hibernate-validator-dev" >> "$GITHUB_OUTPUT"
+              echo "topic=GitHub workflow failures" >> "$GITHUB_OUTPUT"
+            ;;
+            "hibernate/hibernate-search")
+              echo "stream=hibernate-search-dev" >> "$GITHUB_OUTPUT"
+              echo "topic=GitHub workflow failures" >> "$GITHUB_OUTPUT"
+            ;;
+            "hibernate/hibernate-orm")
+              echo "stream=hibernate-orm-dev" >> "$GITHUB_OUTPUT"
+              echo "topic=GitHub workflow failures" >> "$GITHUB_OUTPUT"
+            ;;
+            "hibernate/hibernate-reactive")
+              echo "stream=hibernate-reactive-dev" >> "$GITHUB_OUTPUT"
+              echo "topic=GitHub workflow failures" >> "$GITHUB_OUTPUT"
+            ;;
+            * )
+            # well... just post to infra and hope for the best :) ?
+              if [[ "$REPO_BRANCH_1" =~ "^hibernate/.*$" ]]; then
+                echo "stream=hibernate-infra" >> "$GITHUB_OUTPUT"
+                echo "topic=GitHub workflow failures" >> "$GITHUB_OUTPUT"
+              else
+                echo "SKIP: '${{ github.repository }}' is not in the Hibernate organization. No notifications will be sent."
+                echo "stream=" >> "$GITHUB_OUTPUT"
+                echo "topic=" >> "$GITHUB_OUTPUT"
+              fi
+            ;;
+          esac
+          echo "message=The GitHub [build #${{github.run_id}}](https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}) failed and requires your attention :warning:" >> "$GITHUB_OUTPUT"
+      - name: Notify the build failure
+        uses: zulip/github-actions-zulip/send-message@e4c8f27c732ba9bd98ac6be0583096dea82feea5 # v1.0.2
+        if: ${{ steps.notification-destination.outputs.stream != '' }}
+        with:
+          api-key: ${{ secrets.ZULIP_GITHUB_WORKFLOW_NOTIFICATION_API_KEY }}
+          email: ${{ secrets.ZULIP_GITHUB_WORKFLOW_NOTIFICATION_EMAIL }}
```

Comment:
I think secrets need to be passed from the calling workflow, correct?

Reply:
yeah ... that's what I wasn't entirely sure about, I mean when I tested the action option it was clearly failing with a message that it cannot access secrets and from the docs, one need to pass secrets to action through input paramteres... so that's when I switched to testing the workflow... here it does not complain about the use of secret in the workflow, and I was hoping that if we have the secret in the repo where the workflow is or as an org secret thigns would work.... because if we end up passing the secret at the "call site" then it kind of defeats the purpose of sharing this particular workflow 😔 😕 🙂